iwyu_ast_util.cc

```diff
@@ -711,17 +711,14 @@ static map<const Type*, const Type*> GetTplTypeResugarMapForFunctionNoCallExpr(
 static map<const Type*, const Type*>
 GetTplTypeResugarMapForFunctionExplicitTplArgs(
     const FunctionDecl* decl,
-    const ASTTemplateArgumentListInfo* explicit_tpl_list) {
+    const TemplateArgumentListInfo& explicit_tpl_list) {
   map<const Type*, const Type*> retval;
-  if (explicit_tpl_list) {
-    for (unsigned i = 0; i < explicit_tpl_list->NumTemplateArgs; ++i) {
-      const TemplateArgument& arg
-          = explicit_tpl_list->getTemplateArgs()[i].getArgument();
-      if (const Type* arg_type = GetTemplateArgAsType(arg)) {
-        retval[GetCanonicalType(arg_type)] = arg_type;
-        VERRS(6) << "Adding an explicit template-function type of interest: "
-                 << PrintableType(arg_type) << "\n";
-      }
+  for (unsigned i = 0; i < explicit_tpl_list.size(); ++i) {
```

Comment:
You should be able to convert this to a range loop like this:

    for (const TemplateArgumentLoc& loc : explicit_tpl_list.arguments()) {
        if (const Type* arg = GetTemplateArgAsType(loc.getArgument()) {
            ...
        }
    }